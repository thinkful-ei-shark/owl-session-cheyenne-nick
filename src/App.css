.App .group-row {
    display: flex;
    flex-direction: row;
}
.App .group-column {
    display: flex;
    flex-direction: column;
}

.App .item {
    flex: 1;
}
.App .three-column {
    flex: 3;
}

.App .hidden {
    display: none;
}
