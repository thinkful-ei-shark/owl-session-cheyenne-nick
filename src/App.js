import React, { Component } from 'react';
import SideBar from './SideBar';
import TheStage from './TheStage';
import BottomBar from './BottomBar';
import './App.css'

class App extends Component {
  render() {
    let participantsOnStage = this.props.participants.filter(participant => {
      if (participant.onStage) return participant;
    });
    return (
      <main className='App'>
        <article className="group-column">
          <div className="item group-row">
            <SideBar participants={this.props.participants} chatEvents={this.props.chatEvents}/>
            <TheStage participants={participantsOnStage} avatar={this.props.user.avatar} 
              name={this.props.user.name}/>
          </div>
          <BottomBar />
        </article>
      </main>
    )
  }
}

export default App;