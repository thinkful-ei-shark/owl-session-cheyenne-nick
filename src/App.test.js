import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import store from './STORE';

describe('<App />', () => {
  it('renders without crashing', () => {
      // Create a DOM element to render the component into
      const div = document.createElement('div');

      // Render the component
      // If something is wrong it will fail here
      ReactDOM.render(<App participants={store.participants} chatEvents={store.chatEvents}
        user={{name:'name', avatar:'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1'}}/>, div);

      // Clean up
      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {

    const app = renderer
        .create(<App participants={store.participants} chatEvents={store.chatEvents}
          user={{name:'name', avatar:'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1'}}/>)
        .toJSON();
    
    expect(app).toMatchSnapshot();
  });
});
