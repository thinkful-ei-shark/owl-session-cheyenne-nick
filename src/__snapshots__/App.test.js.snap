// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the UI as expected 1`] = `
<main
  className="App"
>
  <article
    className="group-column"
  >
    <div
      className="item group-row"
    >
      <section
        className="side-bar item group-column"
      >
        <div
          className="actions-header group-row"
        >
          <button
            className="item"
            type="button"
          >
            Chat
          </button>
          <button
            className="item"
            type="button"
          >
            Participants 
            5
          </button>
          <button
            className="item"
            type="button"
          >
            ‚öôÔ∏è
          </button>
          <button
            className="item"
            type="button"
          >
            üîó
          </button>
          <button
            className="item"
            type="button"
          >
            ?
          </button>
        </div>
        <ul
          className="chat-log item"
        >
          <li
            className="side-bar-person group-row"
          >
            <img
              alt="Axe Kubicka"
              className="profile-pic item"
              src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
            />
            <div
              className="person-data group-vertical item"
            >
              <div
                className="group-row"
              >
                <h3
                  className="item"
                >
                  this.props.name
                </h3>
                <p
                  className="item"
                >
                  this.props.time
                </p>
              </div>
              <div
                className="person-data-content group-vertical"
              >
                <p
                  className="item"
                >
                  Hello world
                </p>
              </div>
            </div>
          </li>
          <p
            className="chat-event item"
          >
            <span
              className="user-name"
            >
              Caty Flucker
            </span>
             gave a thumbs up üëç
          </p>
          <p
            className="chat-event item"
          >
            <span
              className="user-name"
            >
              Frank Runciman
            </span>
             gave a thumbs down üëé
          </p>
          <p
            className="chat-event item"
          >
            <span
              className="user-name"
            >
              Koren Templeton
            </span>
             raised their hand ‚úã
          </p>
          <p
            className="chat-event item"
          >
            <span
              className="user-name"
            >
              Koren Templeton
            </span>
             clapped üëè
          </p>
          <p
            className="chat-event item"
          >
            <span
              className="user-name"
            >
              Koren Templeton
            </span>
             joined
          </p>
          <p
            className="chat-event item"
          >
            <span
              className="user-name"
            >
              Ashla Attwool
            </span>
             left
          </p>
          <p
            className="chat-event item"
          >
            <span
              className="user-name"
            >
              Axe Kubicka
            </span>
             joined the stage
          </p>
          <p
            className="chat-event item"
          >
            <span
              className="user-name"
            >
              Axe Kubicka
            </span>
             left the stage
          </p>
        </ul>
        <ul
          className="participant-list hidden item"
        >
          <li
            className="side-bar-person group-row"
          >
            <img
              alt="Koren Templeton"
              className="profile-pic item"
              src={
                Array [
                  "https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1",
                ]
              }
            />
            <div
              className="person-data group-vertical item"
            >
              <h3
                className="item"
              >
                this.props.name
              </h3>
              <div
                className="person-data-content group-row"
              >
                <div
                  className="circle item"
                />
                <p
                  className="stage-status"
                >
                  on stage
                </p>
                <button
                  className="boot"
                >
                  ‚Ä¢ Boot
                </button>
              </div>
            </div>
          </li>
          <li
            className="side-bar-person group-row"
          >
            <img
              alt="Caty Flucker"
              className="profile-pic item"
              src={
                Array [
                  "https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1",
                ]
              }
            />
            <div
              className="person-data group-vertical item"
            >
              <h3
                className="item"
              >
                this.props.name
              </h3>
              <div
                className="person-data-content group-row"
              >
                <div
                  className="circle item"
                />
                
                <button
                  className="boot"
                >
                  ‚Ä¢ Boot
                </button>
              </div>
            </div>
          </li>
          <li
            className="side-bar-person group-row"
          >
            <img
              alt="Frank Runciman"
              className="profile-pic item"
              src={
                Array [
                  "https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1",
                ]
              }
            />
            <div
              className="person-data group-vertical item"
            >
              <h3
                className="item"
              >
                this.props.name
              </h3>
              <div
                className="person-data-content group-row"
              >
                <div
                  className="circle item"
                />
                
                <button
                  className="boot"
                >
                  ‚Ä¢ Boot
                </button>
              </div>
            </div>
          </li>
          <li
            className="side-bar-person group-row"
          >
            <img
              alt="Ashla Attwool"
              className="profile-pic item"
              src={
                Array [
                  "https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1",
                ]
              }
            />
            <div
              className="person-data group-vertical item"
            >
              <h3
                className="item"
              >
                this.props.name
              </h3>
              <div
                className="person-data-content group-row"
              >
                <div
                  className="circle item"
                />
                <p
                  className="stage-status"
                >
                  on stage
                </p>
                <button
                  className="boot"
                >
                  ‚Ä¢ Boot
                </button>
              </div>
            </div>
          </li>
        </ul>
        <textarea
          placeholder="Chat"
        />
      </section>
      <div
        className="the-stage three-column"
      >
        <div
          className="stage-paricipants group-row"
        >
          <div
            className="stage-participant item group-column"
          >
            <div
              className="stage-settings item group-column"
            >
              <h3
                className="stage-name"
              >
                this.props.name
              </h3>
              <div
                className="audio-bar"
              />
              <div
                className="icons group-row"
              >
                <button
                  className="speaker"
                >
                  üîä
                </button>
                <button
                  className="mag-glass"
                >
                  üîç
                </button>
              </div>
            </div>
            <img
              alt="Koren Templeton"
              className="profile-pic item"
              src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
            />
          </div>
          <div
            className="stage-participant item group-column"
          >
            <div
              className="stage-settings item group-column"
            >
              <h3
                className="stage-name"
              >
                this.props.name
              </h3>
              <div
                className="audio-bar"
              />
              <div
                className="icons group-row"
              >
                <button
                  className="speaker"
                >
                  üîä
                </button>
                <button
                  className="mag-glass"
                >
                  üîç
                </button>
              </div>
            </div>
            <img
              alt="Ashla Attwool"
              className="profile-pic item"
              src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
            />
          </div>
          <div
            className="stage-screen"
          >
            <div
              className="audio-bar"
            />
            <img
              alt="name"
              className="profile-pic item"
              src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
            />
          </div>
        </div>
      </div>
    </div>
    <section
      className="bottom-bar item group-row"
    >
      <div
        className="actions-header group-row item"
      >
        <button
          className="item"
          type="button"
        >
          &lt;
        </button>
        <button
          className="item"
          type="button"
        >
          üëç
        </button>
        <button
          className="item"
          type="button"
        >
          üëé
        </button>
        <button
          className="item"
          type="button"
        >
          ‚úã
        </button>
        <button
          className="item"
          type="button"
        >
          üëè
        </button>
      </div>
      <div
        className="actions-header group-row item"
      >
        <button
          className="item"
          type="button"
        >
          üìπ
        </button>
        <button
          className="item"
          type="button"
        >
          üîä
        </button>
        <button
          className="item"
          type="button"
        >
          üíª
        </button>
        <button
          className="item"
          type="button"
        >
          Leave stage
        </button>
      </div>
    </section>
  </article>
</main>
`;
